### ISSUE TITLE
Logitech Gamepad F310 trigger buttons not being detected

### ISSUE THREAD

**Issue by kozross** [ValveSoftware/steam-for-linux#7296]
#### Your system information

* Steam client version (build number or date): Jul 16 2020
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

In at least two games (namely Slay the Spire and Othercide), my Logitech Gamepad F310's left and right trigger buttons are not being detected. This ranges from annoying (Slay the Spire) to basically rendering the game unplayable (Othercide).

I can confirm that my controller is detected by my OS, and that the trigger buttons register when pressed (as confirmed by `jstest`).

#### Steps for reproducing this issue:

1. Plug in controller set to XInput mode
2. Start Slay the Spire
3. Choose a new game
4. Try using left trigger to set the seed
5. Observe nothing happening

**Comment by kozross** [comment]
I can add more evidence to the pile that this is a Steam issue - another game I have (running under Wine) detects the trigger buttons fine on the same controller.

**Comment by kozross** [comment]
Any progress on this issue @kisak-valve ? I am happy to supply additional information if it would aid a speedy resolution.

**Comment by Nobutarou** [comment]
Hi, 
I have F310 but play STS via GOG. I also meet the issue. I don't think this is a Steam issue.

I think the controller mapping file,  game/betaPreferences/STSInputSettings_Controller is wrong,

As the Linux kernel default behavior of xpad module, the left trigger is act as axis-2 and the right trigger is axis-5. The right stick up/down is axis-3, and the left/light is axis-4. The STS default wants axis-3 as left/right triggers. 

Then I modified the file to the following and now I can use the left/right trigger in my play,
```
{
  "SELECT": "0",
  "CANCEL": "1",
  "VIEW": "2",
  "PROCEED": "3",
  "PEEK": "8",
  "PAGE_LEFT_KEY": "4",
  "PAGE_RIGHT_KEY": "5",
  "MAP": "6",
  "SETTINGS": "7",
  "LS_Y_POSITIVE": "-1000",
  "LS_Y_NEGATIVE": "1000",
  "LS_X_NEGATIVE": "-1001",
  "LS_X_POSITIVE": "1001",
  "RS_Y_POSITIVE": "1002",
  "RS_Y_NEGATIVE": "-1002",
  "RS_X_POSITIVE": "-1003",
  "RS_X_NEGATIVE": "1003",
  "DRAW_PILE": "1002",
  "DISCARD_PILE": "1005"
}
```

**Comment by kozross** [comment]
@Nobutarou Yeah, it does seem game-specific. Have you reported this to the dev team for StS?

**Comment by Nobutarou** [comment]
Hi,
No, I didn't. I hesitate to report the issue since I don't use any
supported Linux distribution.
Thanks,
Nobuyuki

2020年10月20日(火) 1:28 Koz Ross <notifications@github.com>:

**Comment by kisak-valve** [comment]
Hello, per "Fixed Logitech F310 controller input on macOS and Linux" in the [2023-03-01](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3650767759928418054) Steam client update, please opt into Steam's client beta and retest this issue.

**Comment by kozross** [comment]
@kisak-valve - just tested Slay the Spire again, using the beta client. Same exact issue.
